microsoft/testsuites/kselftest/kselftest.py

```diff
@@ -0,0 +1,57 @@
+from pathlib import PurePath, PurePosixPath
+from dataclasses import dataclass
+from lisa.executable import Tool
+from lisa.messages import TestStatus
+from typing import Any
+from lisa.tools.remote_copy import RemoteCopy
+from lisa.tools.tar import Tar
+import os
+from lisa.util import LisaException
+
+@dataclass
+class KselftestResult:
+    version: str = ""
+    name: str = ""
+    status: TestStatus = TestStatus.QUEUED
+    exit_value: int = 0
+
+class Kselftest(Tool):
+    @property
+    def command(self) -> str:
+        return str(self._command)
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+        # command to run kselftests
+        self._command = self.get_tool_path(use_global=True) / "kselftest-packages/run_kselftest.sh > kselftest-results.txt"
+        # tar file path specified in yml
+        self._tar_file_path = kwargs.pop("kselftest_file_path", "")
+
+        if not self._tar_file_path:
+            raise LisaException("tar file not found")
+
+    def _install(self) -> bool:
+        # tool_path corresponds to path /home/lisatest/lisa_working/tool/kselftest
+        tool_path = self.get_tool_path(use_global=True)
+        # file_name corresponds to /home/lisatest/lisa_working/tool/kselftest/kselftests.tar.xz
+        file_name = tool_path/os.path.basename(self._tar_file_path)
+        # copies tar file from local machine to remote machine
+        self.node.shell.copy(self._tar_file_path, file_name)
+
+        self.node.tools[Tar].extract(file_name, tool_path)
+        self._log.debug(f"INSTALL from {self._tar_file_path}!")
+
+        return self._check_exists()
+
+    def run_all(self, log_path: str,) -> bool:
+        self.run("", force_run=True, sudo=True, shell=True)
+        # copy kselftest-results.txt from remote to local node for processing results
+        remote_copy = self.node.tools[RemoteCopy]
+        remote_copy.copy_to_local(
```

Comment:
In next step, the results should be parsed as sub test results. So it's can be collected in subtests reports.